pkg/registry/apis/folders/hooks.go

```diff
@@ -66,6 +68,22 @@ func (b *FolderAPIBuilder) beginUpdate(ctx context.Context, obj runtime.Object,
 	}, nil
 }
 
+func (b *FolderAPIBuilder) afterDelete(obj runtime.Object, _ *metav1.DeleteOptions) {
+	ctx := context.Background()
+	log := logging.FromContext(ctx)
+	meta, err := utils.MetaAccessor(obj)
+	if err != nil {
+		log.Error("Failed to access deleted folder object metadata", "error", err)
+		return
+	}
+
+	log.Info("Propagating deleted folder to Zanzana", "folder", meta.GetName(), "parent", meta.GetFolder())
+	err = b.permissionStore.DeleteFolderParents(ctx, meta.GetNamespace(), meta.GetName())
+	if err != nil {
+		logging.FromContext(ctx).Warn("failed to propagate folder to zanzana", "err", err)
```

Comment:
Why don't you use `log` here?

Reply:
Good call, copy/paste miss.  Fixed here: https://github.com/grafana/grafana/pull/111649/commits/c3e365a91fc7ad9609bbf8816d2ec1d5334b42bd